src/duckling/canvas/tools/entity-move-tool.ts

```diff
@@ -75,43 +95,137 @@ export class EntityMoveTool extends BaseTool {
         this._cancel();
     }
 
-    get key() {
-        return "EntityMoveTool";
+    private _processCornerResize(currentCoords : Vector) {
+        let selectedEntityKey = this._selectionService.selection.value.selectedEntity;
```

Comment:
You should extract as much code as you can into stateless functions in another file. Consider naming it something like selection-box.ts